File: java1/src/main/java/toolgood/algorithm/litJson/ParserToken.java
package toolgood.algorithm.litJson;

public enum ParserToken {
    None(0x10000),
    Number(0x10001),
    True(0x10002),
    False(0x10003),
    Null(0x10004),
    CharSeq(0x10005),
    // Single char
    Char(0x10006),

    // Parser Rules (see section A.2.1 of the manual)
    Text(0x10007),
    Object(0x10008),
    ObjectPrime(0x10009),
    Pair(0x10010),
    PairRest(0x10011),
    Array(0x10012),
    ArrayPrime(0x10013),
    Value(0x10014),
    ValueRest(0x10015),
    String(0x10016),

    // End of input
    End(0x10017),

    // The empty rule
    Epsilon(0x10018);


    public int value;  
    // 构造方法  
    private ParserToken( int index) {  
         this.value = index;  
    }  
}

File: JavaScript/src/LitJson/ParserToken.js
let ParserToken = {
    // Lexer tokens (see section A.1.1. of the manual)
    none: 65536, // System.Char.MaxValue + 1

    number: 65537,
    true: 65538,
    false: 65539,
    null: 65540,
    charSeq: 65541,

    // Single char
    char: 65542,

    // Parser Rules (see section A.2.1 of the manual)
    text: 65543,

    object: 65544,
    objectPrime: 65545,
    pair: 65546,
    pairRest: 65547,
    array: 65548,
    arrayPrime: 65549,
    value: 65550,
    valueRest: 65551,
    string: 65552,

    // End of input
    end: 65553,

    // The empty rule
    epsilon: 65554
};

export { ParserToken };